packages/blueprints/blueprint/src/ui-selectors/options-schemas.ts

```diff
@@ -0,0 +1,43 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import { Component } from 'projen';
+import { DynamicKVInput } from './dynamic-kv-input';
+import { Blueprint } from '../blueprint';
+
+/**
+ * This represents an options schema based on DynamicKVInput types
+ */
+export type KVSchema = DynamicKVInput[];
+
+/**
+ * This represents all wizard supported schemas.
+ */
+export type OptionsSchemaType = KVSchema;
+
+/**
+ * This represents a section of the wizard that can be dynamically rendered from a schema provided by the synthesis process.
+ */
+export type OptionsSchemaDefinition<
+  // @ts-ignore:next-line value is never read
+  Identifier extends string,
+  // @ts-ignore:next-line value is never read
+  SchemaType extends OptionsSchemaType,
+  ReturnType extends any = any,
+> = ReturnType | undefined;
+
+/**
+ * This component can be used to define a schema with a given identifier.
+ */
+export class OptionsSchema<T extends OptionsSchemaType> extends Component {
```

Comment:
Not sure we actually want this as a projen component. In this case it will only get written after the synthesis is executed. We probably want to write it immediately, it may make sense to just fs.write it. 

This can be limiting. For example, it stops us from reading the file on the fly as the synthesis process is executing.

Reply:
class signature probably stays the same, but no need to do any of the other bits of it.

Reply:
> This can be limiting. For example, it stops us from reading the file on the fly as the synthesis process is executing.

What is the use case for wanting to read these?  They get written to the workspace like any other resource, and surfaced through preview.

Reply:
Yeah but they can be written to the workspace immediately if they arent projen constructs. This is advantageous because we dont have to wait for the synthesis to end to get them

Reply:
We discussed the following points:
- It might be advantageous to make options available quickly for synthesis tasks that take a long time.
- One advantage of trying it to synthesis is that we know when it completes, so we know when our current options are reflected in the output.  Without some indicator, we can't know if the options reflect our changes.
- Authors can possibly short circuit updates.  This works whether the class is a projen component or not.